From 9560f840f301868b4b4b175e81afa96dc57611ac Mon Sep 17 00:00:00 2001
From: Arnd Bergmann <arnd@arndb.de>
Date: Tue, 6 Nov 2012 23:14:49 +0100
Subject: [PATCH 500/951] ARM: socfpga: mark secondary_trampoline as cpuinit

The socfpga_boot_secondary is correctly marked as __cpuinit but references
secondary_trampoline, which was introduced into the init section, causing
this build warning:

WARNING: vmlinux.o(.cpuinit.text+0xf90): Section mismatch in reference from the function socfpga_boot_secondary() to the variable .init.text:secondary_trampoline
The function __cpuinit socfpga_boot_secondary() references
a variable __init secondary_trampoline.
If secondary_trampoline is only used by socfpga_boot_secondary then
annotate secondary_trampoline with a matching annotation.

The solution is to mark secondary_trampoline as cpuinit as well.

Signed-off-by: Arnd Bergmann <arnd@arndb.de>
---
 arch/arm/mach-socfpga/headsmp.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-socfpga/headsmp.S b/arch/arm/mach-socfpga/headsmp.S
index 65762ab..f09b128 100644
--- a/arch/arm/mach-socfpga/headsmp.S
+++ b/arch/arm/mach-socfpga/headsmp.S
@@ -10,7 +10,7 @@
 #include <linux/linkage.h>
 #include <linux/init.h>
 
-	__INIT
+	__CPUINIT
 	.arch	armv7-a
 
 #define CPU1_START_ADDR 	        0xffd08010
-- 
1.8.0

